src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +121,33 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            return SkipLastIterator(source, Math.Max(count, 0));
+        }
+
+        private static IEnumerable<TSource> SkipLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        {
+            var queue = new Queue<TSource>();
```

Comment:
@JonHanna @ViIvanov In the future I am planning to use a `LargeArrayBuilder` here, which since https://github.com/dotnet/corefx/pull/14020 supporting limiting how many elements are allocated. Then we can call `ToArray()` on it and wrap a struct circular buffer around the resulting array.

@ViIvanov 

> Also, in my practice, `items.SkipLast(1)` used often and we can use single TSource instead of `new Queue<TSource>` in this case.

👍 Noted.